File: 프로그래머스/Level1/[1차] 다트 게임.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(string dartResult) {
    int answer = 0;
    int i=0;
    vector <int> arr;
    
    for(int j=0; j<dartResult.length(); j++){
        
        if(dartResult[j]=='D'){
            arr[i-1] *= arr[i-1];
        }
        else if(dartResult[j]=='T'){
            arr[i-1] = arr[i-1]*arr[i-1]*arr[i-1];
        }
        else if(dartResult[j]=='S'){
            
        }
        else if(dartResult[j]=='*'){
            arr[i-1] *= 2;
            if(i>1){
                arr[i-2] *= 2;
            }
        }
        else if(dartResult[j]=='#'){
            arr[i-1] = arr[i-1] * -1;
        }
        else{
            if(dartResult[j+1]=='0'){
                arr.push_back(10);
                j++;
            }
            else
                arr.push_back(dartResult[j]-'0');
            i++;
        }
            
    }
    
        
    for(int i=0; i< arr.size(); i++){
        answer += arr[i];
    }
    
    return answer;
}

File: 프로그래머스/Level1/[1차] 다트 게임.js
function solution(dartResult) {
    let now =0, before = 0, k = 0;
    let answer = 0;
    
    for(let i=0; i<dartResult.length; i++){
        if(dartResult[i] === '1' && dartResult[i+1] === '0'){
            k = 10;
            i++;
            continue;
        }
        else if(dartResult[i] <= '9' && dartResult[i] >= '0'){
            k = parseInt(dartResult[i], 10);
            continue;
        }
        
        if(dartResult[i] === 'S'){
            now = k;
            answer += now;
        }
        else if(dartResult[i] === 'D'){
            now = Math.pow(k, 2);
            answer += now;
        }
        else if(dartResult[i] === 'T'){
            now = Math.pow(k, 3);
            answer += now;
        }        
        
        if(dartResult[i+1] === '*'){
            answer += (now + before);
            now *= 2;
            i++;
        }
        else if(dartResult[i+1] === '#'){
            answer -= now;
            now = now * (-1);
            answer += now;
            i++;
        }
        
        before = now;
    }
    return answer;
}
